src/auto-advance.js

```diff
@@ -1,97 +1,97 @@
+ * Manages the auto-advance functionality in a media player.
+ * Auto-advance automatically moves to the next item after a specified delay.
  */
-let reset = (player) => {
-  const aa = player.playlist.autoadvance_;
+export default class AutoAdvance {
 
-  if (aa.timeout) {
-    player.clearTimeout(aa.timeout);
+  /**
+   * Creates an instance of the AutoAdvance class.
+   *
+   * @param {Object} player
+   *        The media player instance.
+   * @param {Function} advanceCallback
+   *        The callback function to execute when advancing to the next item.
+   */
+  constructor(player, advanceCallback) {
+    this.player_ = player;
+    this.advanceCallback_ = advanceCallback;
+    this.delay_ = null;
+    this.timeoutId_ = null;
+    this.startTimeout_ = this.startTimeout_.bind(this);
   }
 
-  if (aa.trigger) {
-    player.off('ended', aa.trigger);
-  }
-
-  aa.timeout = null;
-  aa.trigger = null;
-};
+  /**
+   * Sets the delay for auto-advance.
+   * If the delay is invalid or not zero or a positive number, auto-advance is cancelled.
+   *
+   * @param {number} seconds
+   *        The delay in seconds before auto-advancing.
+   */
+  setDelay(seconds) {
+    // Cancel any existing auto-advance behavior and start fresh
+    this.fullReset();
 
-/**
- * Sets up auto-advance behavior on a player.
- *
- * @param  {Player} player
- *         the current player
- *
- * @param  {number} delay
- *         The number of seconds to wait before each auto-advance.
- *
- * @return {undefined}
- *         Used to short circuit function logic
- */
-const setup = (player, delay) => {
-  reset(player);
+    // If delay is invalid or undefined, do nothing further (auto-advance already cancelled)
+    if (typeof seconds !== 'number' || seconds < 0 || !isFinite(seconds)) {
+      return;
+    }
 
-  // Before queuing up new auto-advance behavior, check if `seconds` was
-  // called with a valid value.
-  if (!validSeconds(delay)) {
-    player.playlist.autoadvance_.delay = null;
-    return;
+    // Set the new delay and start listening for 'ended' to trigger auto-advance
+    this.delay_ = seconds;
+    this.player_.on('ended', this.startTimeout_);
   }
 
-  player.playlist.autoadvance_.delay = delay;
+  /**
+   * Gets the delay for auto-advance.
+   */
+  getDelay() {
+    return this.delay_;
+  }
 
-  player.playlist.autoadvance_.trigger = function() {
+  /**
+   * Starts the auto-advance timeout.
+   *
+   * @private
+   */
+  startTimeout_() {
+    // Ensure we don't stack timeouts
+    this.clearTimeout_();
 
-    // This calls setup again, which will reset the existing auto-advance and
-    // set up another auto-advance for the next "ended" event.
-    const cancelOnPlay = () => setup(player, delay);
+    // Set a new timeout for auto-advancing
+    if (this.delay_ !== null) {
+      // Listen for a play event to cancel auto-advance if it occurs before the timeout completes
+      this.clearTimeoutOnPlay = () => this.clearTimeout_();
```

Comment:
I think it should not be public...
Also, I think we should get rid of it and use auto-bind for `clearTimeout_`.